.github/workflows/tests.yml

```diff
@@ -366,6 +366,8 @@ jobs:
       # Build initial Synapse image
       - run: docker build -t matrixdotorg/synapse:latest -f docker/Dockerfile .
         working-directory: synapse
+        env:
+          DOCKER_BUILDKIT: 1
```

Comment:
Am I right in thinking that this doesn't make the `tests` workflow faster; but it ensures the built image has support for this caching setup?

Reply:
neither, I'm afraid. (The idea is not to change anything about the built image; it is all about the process of building that image).

This is needed because otherwise the `docker build` command will complain about not understanding the `--mount` flags in the Dockerfile. Those flags are unlikely to do much in this workflow, but they are super helpful when working locally, hence wanting them to be there.